### ISSUE TITLE
Internal Server Error when completing challenge / getting advancement

### ISSUE THREAD

**Issue by Azahliil** [arthurbambou/Fabric---Discord-Link#98]
I noticed that I get kicked every time after getting an advancement (technically) with the message "internal server error" in the client.

The error message from the logs is as follows:
```14:12:26 CONSOLE: [INFO] Azahliil has completed the challenge [Cover Me in Debris]
14:12:26 CONSOLE: thread/WARN]: Failed to handle packet for /XX.XX.XX.X:XXXXX
14:12:26 CONSOLE: java.lang.ClassCastException: net.minecraft.class_2585 cannot be cast to net.minecraft.class_2588
14:12:26 CONSOLE: Azahliil has left.
```

My guess (and it's just a guess, it might be caused by something completely different) is that this has something to do with fdlink, as the advancemnets don't get announced in discord.

Server:
- fabric 0.9.1 build.205

Mods:
- lithium 1.16.1-0.51
- FDLink 0.5.1
- fabric api build.385

My FDLink Config:

```json
{
  "token": "XXXXX",
  "chatChannels": [
    "XXXXX"
  ],
  "logChannels": [
    "XXXXX"
  ],
  "ignoreBots": true,
  "minecraftToDiscord": {
    "general": {
      "enableDebugLogs": false
    },
    "messages": {
      "playerMessage": {
        "customMessage": "\u003c%player\u003e %message",
        "useCustomMessage": true
      },
      "serverStarting": "**Server is starting !**",
      "serverStarted": "**Server Started**",
      "serverStopping": "**Server is stopping!**",
      "serverStopped": "**Server Stopped**",
      "playerJoined": {
        "customMessage": "**%player joined the game**",
        "useCustomMessage": true
      },
      "playerJoinedRenamed": {
        "customMessage": "**%new (formerly known as %old) joined the game**",
        "useCustomMessage": true
      },
      "playerLeft": {
        "customMessage": "**%player left the game**",
        "useCustomMessage": true
      },
      "advancementTask": {
        "customMessage": "**%player has made the advancement %advancement**",
        "useCustomMessage": true
      },
      "advancementChallenge": {
        "customMessage": "**%player has completed the challenge %advancement**",
        "useCustomMessage": true
      },
      "advancementGoal": {
        "customMessage": "**%player has reached the goal %advancement**",
        "useCustomMessage": true
      },
      "deathMsgPrefix": ":skull_crossbones: **",
      "deathMsgPostfix": "**"
    },
    "chatChannels": {
      "commandPrefix": "-",
      "allowDiscordCommands": true,
      "serverStartingMessage": true,
      "serverStartMessage": true,
      "serverStopMessage": true,
      "serverStoppingMessage": true,
      "customChannelDescription": false,
      "minecraftToDiscordTag": false,
      "minecraftToDiscordDiscriminator": false,
      "playerMessages": true,
      "joinAndLeaveMessages": true,
      "advancementMessages": true,
      "challengeMessages": true,
      "goalMessages": true,
      "deathMessages": true,
      "sendMeCommand": false,
      "sendSayCommand": true,
      "adminMessages": false
    },
    "logChannels": {
      "serverStartingMessage": true,
      "serverStartMessage": true,
      "serverStopMessage": true,
      "serverStoppingMessage": true,
      "customChannelDescription": false,
      "minecraftToDiscordTag": false,
      "minecraftToDiscordDiscriminator": false,
      "playerMessages": false,
      "joinAndLeaveMessages": true,
      "advancementMessages": false,
      "challengeMessages": false,
      "goalMessages": false,
      "deathMessages": true,
      "sendMeCommand": true,
      "sendSayCommand": true,
      "adminMessages": true
    }
  },
  "discordToMinecraft": {
    "pingLongVersion": false,
    "message": "[%player] %message",
    "commandPrefix": "!"
  },
  "emojiMap": [
    {
      "name": "example_name",
      "id": ":example_id:22222222"
    },
    {
      "name": "example_name2",
      "id": ":example_id2:22222222"
    }
  ]
}
```

The log channel is on a different discord server if that makes any difference

Thank you so much in advance :3

**Comment by MightyKnight** [comment]
@Azahliil Fixed in [0.5.2](https://www.curseforge.com/minecraft/mc-mods/fabric-discord-link/files/3026982), thanks for reporting.

**Comment by Azahliil** [comment]
Thank you so much for the quick fix!
